src/client/Microsoft.Identity.Client/OAuth2/MsalTokenResponse.cs

```diff
@@ -117,6 +113,10 @@ public IReadOnlyDictionary<string, string> CreateExtensionDataStringMap()
                 {
                     stringExtensionData.Add(item.Key, item.Value.ToString());
                 }
+                else if (item.Value.Type == JTokenType.Array || item.Value.Type == JTokenType.Object)
+                {
+                    stringExtensionData.Add(item.Key, item.Value.ToString(Formatting.None));

```

Comment:
What does `Formatting.None` do above for scalars?

Reply:
Not much of an effect but we can keep the logic the same